### PR TITLE
Require source and target fields in tokenization config

### PR DESCRIPTION
**Pull request by guillaumekln**
To make things simpler, source and target fields should always be
set. 'multi' can be used only to configure shared subword and
vocabulary options.

This change also clarifies in the consumer implementation that multi
and source/target are mutually exclusive.

### CODE REVIEW

**Comment by guillaumekln** [comment]
The CodeFactor issue is related to the added TODO comment. It can be ignored.

File: nmtwizard/preprocess/preprocess.py [hunk 1]
```diff
@@ -111,12 +111,11 @@ def _generate_models(self, tokenization_step, option):
         opt_source = tok_config.get('source', {}).get(build_option)
         opt_target = tok_config.get('target', {}).get(build_option)
 
-        if not opt_multi and not (opt_source and opt_target):
+        if not opt_multi and not opt_source and not opt_target:
```
Comment (natsegal) [inline nmtwizard/preprocess/preprocess.py:114]: Not sure why this change. For subword / vocabulary building we want the options to be specified either in multi or in both source and target, don't we ?

**Review by natsegal — APPROVED** [review]
Looks good to me

File: nmtwizard/preprocess/preprocess.py (hunk 1 above)
Comment (guillaumekln) [reply nmtwizard/preprocess/preprocess.py:114]: I think we also need to allow building for just the source, or just the target.

Comment (natsegal) [reply nmtwizard/preprocess/preprocess.py:114]: Ok, it's just a detail.
